src/main/webapp/config.js

```diff
@@ -87,30 +130,33 @@ const populateSiteForm = (index) => {
     const webhookUrl = document.getElementById(`webhookUrl_${index}`).textContent;
     const credentialsId = document.getElementById(`credentialsId_${index}`).textContent;
 
-    // get site form elements
+    setSiteFormContent(siteName, webhookUrl, credentialsId);
+}
+
+const setSiteFormContent = (site = "", webhook = "", credentials = "") => {
     const siteDataContainer = document.getElementById('siteDataContainer');
+
     const siteInput = siteDataContainer.querySelector('[name="site"]');
     const webhookUrlInput = siteDataContainer.querySelector('[name="webhookUrl"]');
     const credentialsInput = siteDataContainer.querySelector('[name="_.credentialsId"]');
 
     // set site form values
-    siteInput.value = siteName;
-    webhookUrlInput.value = webhookUrl;
-    credentialsInput.value = credentialsId;
+    siteInput.value = site;
+    webhookUrlInput.value = webhook;
+    credentialsInput.value = credentials;
+
+    // Run input validation
+    invokeOnChangeChecks(siteInput, webhookUrlInput);
 }
 
 // This forces the serverside validation event after we populate the site form
-const invokeOnChangeChecks = () => {
-    const webHookUrlInput = document.querySelector('[name="webhookUrl"]');
-    const siteInput = document.querySelector('[name="site"]');
-
-    triggerChangeEvent(webHookUrlInput);
-    triggerChangeEvent(siteInput);
-};
-
-const triggerChangeEvent = (element) => {
-    element && element.dispatchEvent(new Event('change'));
-};
+const invokeOnChangeChecks = (...args) => {
```

Comment:
There are multiple ways to achieve the same thing, but Jenkinshas its little claws in stuff everywhere so this approach was the only one that was successful everytime!